lib/matplotlib/ticker.py

```diff
@@ -294,6 +294,13 @@ def _set_locator(self, locator):
         """ Subclasses may want to override this to set a locator. """
         pass
 
+    @cbook.deprecated("3.1")
+    def pprint_val(self, *args, **kwargs):
+        return self._pprint_val(*args, **kwargs)
+
+    def _pprint_val(self, *args, **kwargs):
+        raise NotImplementedError
```

Comment:
Should get a comment, what this is supposd to do and how subclasses should implement it. This info is targeted at other developers.

Reply:
Moved the deprecation to each individual class, so no need for the NotImplemented base class version anymore.